keps/sig-node/3008-cri-class-based-resources/README.md

```diff
@@ -0,0 +1,1561 @@
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+In this section we refer to different
+[implementation phases](#implementation-phases). In this KEP we're now
+targeting phase 1.
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ClassResources
+  - Components depending on the feature gate:
+    - Implementation Phase 1:
+        - kubelet
+    - Future phases (with updated pod spec and scheduler and quota support):
+        - kubelet
+        - kube-apiserver
+        - kube-scheduler
+        - kube-controller-manager
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+Yes it can.
+
+Implementation Phase 1: In this phase pod annotations are used as the user
+interface for assigning QoS-class resources to workloads. Existing
+running workloads continue to work without any changes as their QoS-class
+resource assigment in the runtime is not changed.
+Restarting or re-deploying a workload causes it to lose its QoS-class resource
+assignment as the annotation parsing in kubelet is disabled. In other words,
+the workload is able to run but the QoS-class resource assignment request from
+the user (via pod annotations) is effectively ignored.
+
+Future implementation phases: running workloads continue to work without any
+changes. Restarting or re-deploying a workload causes it to fail as the
+requested QoS-class resources are not available.
```

Comment:
? clarify pls why the resources are not available after restart.. and how the annotations are disabled by kubelet.. filtered out?

Reply:
I reworded this a bit. If featuregate is disabled kubelet will simply ignore the annotations.

Reply:
Marking as resolved as we don't even have annotations in the proposal anymore